src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/Talisman.java

```diff
@@ -130,48 +134,61 @@ private static boolean hasMessage(Talisman talisman) {
         return talisman.getMessageSuffix() != null;
     }
 
-    public static boolean checkFor(Event e, SlimefunItemStack stack) {
+    public static ItemStack checkFor(@Nonnull Event e, @Nonnull SlimefunItemStack stack) {
         return checkFor(e, stack.getItem());
     }
```

Comment:
First off, I would have still left the existing method. And I think these methods can use a bit renaming too.
```suggestion
    public static boolean activate(@Nonnull Event e, @Nonnull SlimefunItemStack stack) {
        return activateAndGet(e, stack) != null;
    }

    public static ItemStack activateAndGet(@Nonnull Event e, @Nonnull SlimefunItemStack stack) {
        return activateAndGet(e, stack.getItem());
    }
```

Reply:
Im going to call the first one tryActivate since it returns a boolean and that makes more sense when reading